llvm/include/llvm/SandboxIR/SandboxIR.h

```diff
@@ -278,12 +381,55 @@ class Context {
   DenseMap<llvm::Value *, std::unique_ptr<sandboxir::Value>>
       LLVMValueToValueMap;
 
+  /// Take ownership of VPtr and store it in `LLVMValueToValueMap`.
+  Value *registerValue(std::unique_ptr<Value> &&VPtr);
+
+  /// Helper for avoiding recursion loop when creating SBConstants.
+  SmallDenseSet<llvm::Constant *, 8> VisitedConstants;
+
+  Value *getOrCreateValueInternal(llvm::Value *V, int Depth,
+                                  llvm::User *U = nullptr);
+
+  template <typename SBT, typename LLVMT>
+  SBT *getOrCreateGeneric(LLVMT *LLVMV) {
```

Comment:
shouldn't `LLVMT` always be a `llvm::Value`?

Reply:
I can drop it from the argument, but I would still need to create an object of LLVMT type, so I don't think I can drop it from the template entirely. It will look like this:
```
 template <typename SBT, typename LLVMT>  
 SBT *getOrCreateGeneric(llvm::Value *LLVMV) {
  ...
      It->second = std::make_unique<SBT>(cast<LLVMT>(LLVMV), *this);
```
Wdyt?

Reply:
is the constructor of SBT always going to take `(LLVMT, Context)` as the arguments? e.g. LLVM Values have different constructor param lists so it doesn't make sense to templatize these sorts of builders in that context (pun intended)

Reply:
OK I will remove the template.